File: app/src/main/java/com/mertcaliskanyurek/binancebot/NotificationBuilder.java
package com.mertcaliskanyurek.binancebot;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

import androidx.annotation.RequiresApi;
import androidx.core.app.NotificationCompat;

import static android.content.Context.NOTIFICATION_SERVICE;

public class NotificationBuilder implements INotification {

    public static final String CHANNEL_NAME = "NotificationChannel";

    private final Context mContext;
    private NotificationManager mNotiManager;

    public NotificationBuilder(Context context){
        this.mContext = context;
        mNotiManager = (NotificationManager) context.getSystemService(NOTIFICATION_SERVICE);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
            initChannel();
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    private void initChannel()
    {
        NotificationChannel serviceChannel = new NotificationChannel(
                CHANNEL_NAME,
                "Binance Service",
                NotificationManager.IMPORTANCE_DEFAULT
        );
        mNotiManager = mContext.getSystemService(NotificationManager.class);
        mNotiManager.createNotificationChannel(serviceChannel);
    }


    @Override
    public Notification buildNotification(String notificationText) {
        Intent notificationIntent = new Intent(mContext, MainActivity.class);
        PendingIntent pendingIntent = PendingIntent.getActivity(mContext,
                0, notificationIntent, 0);

        return new NotificationCompat.Builder(mContext, CHANNEL_NAME)
                .setSmallIcon(R.drawable.ic_launcher_background)
                .setContentTitle("Binance")
                .setContentText(notificationText)
                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                // Set the intent that will fire when the user taps the notification
                .setContentIntent(pendingIntent)
                .setAutoCancel(true).build();
    }

    @Override
    public void notify(int id, Notification notification) {
        mNotiManager.notify(id,notification);
    }
}

interface INotification {
    Notification buildNotification(String notification);
    void notify(int id, Notification notificationBuilder);
}

File: app/src/main/java/com/mertcaliskanyurek/binancebot/NotificationBuilder.kt
package com.mertcaliskanyurek.binancebot

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.annotation.RequiresApi
import androidx.core.app.NotificationCompat

class NotificationBuilder(private val mContext: Context) : INotification {
    private var mNotiManager: NotificationManager

    @RequiresApi(api = Build.VERSION_CODES.O)
    private fun initChannel() {
        val serviceChannel = NotificationChannel(
                CHANNEL_NAME,
                NOTIFICATION_NAME,
                NotificationManager.IMPORTANCE_DEFAULT
        )
        mNotiManager = mContext.getSystemService<NotificationManager>(NotificationManager::class.java)
        mNotiManager.createNotificationChannel(serviceChannel)
    }

    override fun buildNotification(notificationText: String?): Notification {
        val notificationIntent = Intent(mContext, MainActivity::class.java)
        val pendingIntent = PendingIntent.getActivity(mContext,
                0, notificationIntent, 0)
        return NotificationCompat.Builder(mContext, CHANNEL_NAME)
                .setSmallIcon(R.drawable.ic_launcher_background)
                .setContentTitle(CONTENT_TITLE)
                .setContentText(notificationText)
                .setPriority(NotificationCompat.PRIORITY_DEFAULT) // Set the intent that will fire when the user taps the notification
                .setContentIntent(pendingIntent).build()
    }

    override fun notify(id: Int, notification: Notification?) {
        mNotiManager.notify(id, notification)
    }

    companion object {
        const val CHANNEL_NAME = "NotificationChannel"
        const val NOTIFICATION_NAME = "Binance Service"
        const val CONTENT_TITLE = "Binance Bot"
    }

    init {
        mNotiManager = mContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) initChannel()
    }
}

public interface INotification {
    fun buildNotification(notification: String?): Notification
    fun notify(id: Int, notificationBuilder: Notification?)
}